YNAB is forced to refresh when re-showing Help or Referral banner after being hidden

### What happened?

Re-showing the help button or the referral banner requires a refresh due to "Something Went Wrong"

```
rollbar.min.js:1 TypeError: this.features.removeAt is not a function
    at ObserveListener.removeFeature (chrome-extension://opijhballkdcgkjkmaeamckkplgighne/web-accessibles/ynab-toolkit.js:101813:19)
    at HideReferralBanner.removeListeners (chrome-extension://opijhballkdcgkjkmaeamckkplgighne/web-accessibles/ynab-toolkit.js:60690:85)
    at YNABToolkit.destroyFeature (chrome-extension://opijhballkdcgkjkmaeamckkplgighne/web-accessibles/ynab-toolkit.js:58837:21)
    at YNABToolkit.onBackgroundMessage (chrome-extension://opijhballkdcgkjkmaeamckkplgighne/web-accessibles/ynab-toolkit.js:58901:34)
e.<computed> @ rollbar.min.js:1Understand this error
rollbar.min.js:1 Toolkit Error:
  - Feature: unknown
  - Feature Setting: unknown
  - Function: global
  - Message: this.features.removeAt is not a function TypeError: this.features.removeAt is not a function
    at ObserveListener.removeFeature (chrome-extension://opijhballkdcgkjkmaeamckkplgighne/web-accessibles/ynab-toolkit.js:101813:19)
    at HideReferralBanner.removeListeners (chrome-extension://opijhballkdcgkjkmaeamckkplgighne/web-accessibles/ynab-toolkit.js:60690:85)
    at YNABToolkit.destroyFeature (chrome-extension://opijhballkdcgkjkmaeamckkplgighne/web-accessibles/ynab-toolkit.js:58837:21)
    at YNABToolkit.onBackgroundMessage (chrome-extension://opijhballkdcgkjkmaeamckkplgighne/web-accessibles/ynab-toolkit.js:58901:34)
e.<computed> @ rollbar.min.js:1Understand this error
observeListener.js:119 Uncaught TypeError: this.features.removeAt is not a function
    at ObserveListener.removeFeature (observeListener.js:119:19)
    at HideReferralBanner.removeListeners (feature.ts:65:5)
    at YNABToolkit.destroyFeature (ynab-toolkit.tsx:109:13)
    at YNABToolkit.onBackgroundMessage (ynab-toolkit.tsx:185:18)Understand this error
rollbar.min.js:1 Error successfully reported to Rollbar. UUID: 272ca447-cd2c-4735-b976-5722594ef2e5
```

### Steps to Reproduce

1. Open settings
2. Select "Hide Referral Banner"
3. De-select "Hide Referral Banner"

### What browsers are you seeing the problem on?

Chrome v129.0.6668.101

### What OSs are you seeing the problem on?

Windows 11

### What Toolkit Version are you running?

3.15.0

### Screenshots

_No response_

### Setting Export

_No response_

--- a/src/extension/listeners/observeListener.js
+++ b/src/extension/listeners/observeListener.js
@@ -116,7 +116,7 @@ export class ObserveListener {
   }
 
   removeFeature(feature) {
-    this.features.removeAt(this.features.indexOf(feature));
+    this.features.splice(this.features.indexOf(feature), 1);
   }
 
   emitChanges() {

--- a/src/extension/listeners/routeChangeListener.js
+++ b/src/extension/listeners/routeChangeListener.js
@@ -46,6 +46,6 @@ export class RouteChangeListener {
   }
 
   removeFeature(feature) {
-    this.features.removeAt(this.features.indexOf(feature));
+    this.features.splice(this.features.indexOf(feature), 1);
   }
 }

